test/test_common/environment.cc

```diff
@@ -149,6 +149,11 @@ std::string TestEnvironment::temporaryFileSubstitute(const std::string& path,
                                                      const ParamMap& param_map,
                                                      const PortMap& port_map,
                                                      Network::Address::IpVersion version) {
+  // Protobufs should be generated dynamically so not need port substitution.
```

Comment:
I think we can get away with runtime replacements rather than template replacements, at least for the current things we're templatizing.  If I'm wrong "someone" will have to go fix it and delete this comment as part of #1162 and conveniently that someone is likely me :-)